src/hotspot/share/opto/ifnode.cpp

```diff
@@ -980,9 +978,10 @@ bool IfNode::fold_compares_helper(ProjNode* proj, ProjNode* success, ProjNode* f
         return false;
       }
     } else {
-      assert(false, "unhandled lo_test: %d", lo_test);
+      assert(igvn->_worklist.member(in(1)) && in(1)->Value(igvn) != igvn->type(in(1)), "unhandled lo_test: %d", lo_test);
       return false;
     }
+    assert(this_bool->_test.is_less() && !fail->_con, "incorrect test");
```

Comment:
This should lead with "this test was canonicalized" comment? Missed during the move, I think.

Reply:
I also find it a bit weird to even have the assert on this path, as we tested all cases in the if-chain before, and the only path to this assert is through `lt` and `le` -- which is `is_less`? Maybe I am missing something, though.

Reply:
> I also find it a bit weird to even have the assert on this path, as we tested all cases in the if-chain before, and the only path to this assert is through `lt` and `le` -- which is `is_less`? Maybe I am missing something, though.

I kept it because of the !fail->_con part of the assert. I kept the whole assert and move it around because I'm lazy, I guess.

Reply:
That's fine. Keep the comment, though?

Reply:
right